src/index.ts

```diff
@@ -13,49 +13,94 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
+import * as request from 'request';
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
 import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
+
+// Returns a promise which will resolve to true if running on GCE and false
+// otherwise.
+// It is assumed a program is running on GCE if the metadata API can be
+// accessed.
+function onGCE(): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    request.get(
+        {url: metadataAPI, headers: {'Metadata-Flavor': 'Google'}},
+        function(err: Error, response) {
+          err ? resolve(false) : resolve(true);
+        });
+  });
+}
+
+// Returns a promise that resolves to the value of metadata field.
```

Comment:
As @ofrobots mentioned in some other PR, there isn't much need to describe the semantics of returning a promise over and over again as it's pretty idiomatic. I think the documentation here and in all other methods that return a promise can be simplified to something like "Returns a GCE metadata value or fails if there is problem communicating the metadata server or the HTTP request fails otherwise."